### PR TITLE
[18.0][MIG] base_time_window: Migration to 18.0

### PR DESCRIPTION
**Pull request by Kimkhoi3010**
It found only 1 commit to port:
It was ignored, i added them to blacklist: [4e65739](https://github.com/OCA/server-tools/pull/3096/commits/4e6573981cebb79503733c46356af7692e5ddc4c)

### CODE REVIEW

File: base_time_window/models/time_window_mixin.py [hunk 1]
```diff
@@ -31,18 +31,16 @@ def check_window_no_overlaps(self):
...
-                        {
-                            "end_time": self.float_to_time_repr(record.time_window_end),
-                            "start_time": self.float_to_time_repr(
-                                record.time_window_start
-                            ),
-                        }
+                    self.env._(
+                        "%s must be > %s",
+                        self.float_to_time_repr(record.time_window_end),
+                        self.float_to_time_repr(record.time_window_start),
```
Comment (sebalix) [inline base_time_window/models/time_window_mixin.py]: We should keep placeholders as their places could be different depending on the language/translation.

File: base_time_window/models/time_window_mixin.py [hunk 2]
```diff
@@ -79,27 +77,23 @@ def check_window_no_overlaps(self):
...
                 other = self.browse(res[0][0])
                 raise ValidationError(
-                    _("%(record_name)s overlaps %(other_name)s")
-                    % (
-                        {
-                            "record_name": record.display_name,
-                            "other_name": other.display_name,
-                        }
+                    self.env._(
+                        "%s overlaps %s", record.display_name, other.display_name
```
Comment (sebalix) [inline base_time_window/models/time_window_mixin.py]: Same remark

**Comment by TDu** [comment]
@Kimkhoi3010  Thanks for the migration, could you attend the comments, please.

**Review by mmequignon — APPROVED** [review]

**Review by sebalix — APPROVED** [review]

**Pull request merged**